pulp_ostree/tests/functional/conftest.py

```diff
@@ -0,0 +1,105 @@
+    RepositorySyncURL,
+    RemotesOstreeApi,
+)
+
+
+@pytest.fixture()
+def ostree_client(_api_client_set, bindings_cfg):
+    """Fixture to provide a client to Pulp API"""
+    api_client = ApiClient(bindings_cfg)
+    _api_client_set.add(api_client)
+    yield api_client
+    _api_client_set.remove(api_client)
+
+
+@pytest.fixture()
+def ostree_content_refs_api(ostree_client):
+    """Fixture that returns an instance of ContentRefsApi"""
+    return ContentRefsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RepositoriesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_remotes_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RemotesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_versions_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeVersionsApi"""
+    return RepositoriesOstreeVersionsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repository_sync_url(ostree_client):
+    """Fixture that returns an instance of RepositorySyncURL"""
+    return RepositorySyncURL(ostree_client)
+
+
+@pytest.fixture()
+def ostree_content_commits_api(ostree_client):
+    """Fixture that returns an instance of ContentCommitsApi"""
+    return ContentCommitsApi(ostree_client)
+
+
+@pytest.fixture()
+def remote(ostree_remotes_api):
+    """Fixture that creates an ostree Remote and returns the object created"""
+    body = gen_ostree_remote(depth=0, policy="immediate")
+    return ostree_remotes_api.create(body)
+
+
+@pytest.fixture()
+def random_name():
+    """Fixture that returns a dictionary with a random uuid for the `name` key"""
+    return {"name": str(uuid.uuid4())}
+
+
+@pytest.fixture()
+def repo(ostree_repositories_api, random_name):
+    """Fixture that creates an ostree Repository"""
+    return ostree_repositories_api.create(random_name)
+
+
+@pytest.fixture()
+def ostree_repository_version(
+    monitor_task, remote, repo, ostree_repositories_api, ostree_repositories_versions_api
+):
+    """
+    Fixture that syncs a Remote ostree Repository,
+    waits until the sync task completes and returns the
+    created repository version object.
+    """
+    repository_sync_data = RepositorySyncURL(remote=remote.pulp_href)
+    response = ostree_repositories_api.sync(repo.pulp_href, repository_sync_data)
+    repo_version = monitor_task(response.task).created_resources[0]
+
+    return ostree_repositories_versions_api.read(repo_version)
+
+
+@pytest.fixture()
+def ostree_repositories_delete(ostree_repositories_api, repo):
+    """Fixture that deletes an ostree Repository"""
+    return ostree_repositories_api.delete(repo.pulp_href)
+
+
+@pytest.fixture()
+def ostree_remotes_delete(ostree_remotes_api, remote):
+    """Fixture that deletes an ostree Remote"""
+    return ostree_remotes_api.delete(remote.pulp_href)
```

Comment:
With pytest, you do not need to care about cleanups in the old-fashioned way. The procedure is as follows:
```
def repo_fixture():
    repo = setup_repo()
    yield repo
    delete_repo(repo)
```

Thus, by following the logic implemented here: https://github.com/pulp/pulp_gem/blob/4de8c2adf1be973958d95b16f0c4137dab7ba066/pulp_gem/tests/functional/conftest.py#L105-L116, we can clearly get rid of these two cleanup fixtures.